src/prefect/schedules/clocks.py

```diff
@@ -303,7 +310,7 @@ def events(self, after: datetime = None) -> Iterable[ClockEvent]:
         if after_localized.microsecond:
             after_localized = after_localized + timedelta(seconds=1)
 
-        cron = croniter(self.cron, after_localized)
+        cron = croniter(self.cron, after_localized, day_or=self.day_or)
```

Comment:
Looks like mypy is failing due to an inaccurate type definition for `croniter`, can just ignore types here

```suggestion
        cron = croniter(self.cron, after_localized, day_or=self.day_or)  # type: ignore
```